### ISSUE TITLE
"Download Data to the Date" option not working [BUG]

### ISSUE THREAD

**Issue by TGJ64** [AAndyProgram/SCrawler#71]
I admittedly haven't used this feature for a while and only noticed recently, so I'm not sure what version it stopped working, but I tried using it in version 2022.9.13.0_x64 and the following issues occurred:
- Instagram "Stories" and "Tagged" photos/videos all downloaded, ignoring the date limitation.
- "Post" photos/videos did not download for Instagram or Twitter (I didn't try the others), regardless of whether or not there were photos within the date range.

I have a couple other suggestions relating to this feature that aren't bugs so much as possible improvements:
- I think the intended functionality is to download any media posted _later_ than the chosen date, however the phrasing of the option makes it seem like it could instead be to select any media posted _before_ a certain date. I think the former is more useful but could see a benefit to being able to download all Media from a range of dates, rather than just "all before" or "all after".
- It would be helpful to have this as an option in the "Download All" options as well, rather than having it only in the right-click menu for each user.

**Comment by AAndyProgram** [comment]
Personally, I haven't used this feature. This feature was requested by several users and I added it. None of them told me that something was wrong.

>Instagram "Stories" and "Tagged" photos/videos all downloaded, ignoring the date limitation.

It is. The date limit does not apply to tagged posts, saved posts, and stories. Stories usually don't include too many photos and/or videos. I don't know how to limit tagged photos by date, because these are tagged photos. A user can take this photo a year ago and tag you just now. How do you suppose to apply the date limit for tagged posts?

>"Post" photos/videos did not download for Instagram or Twitter (I didn't try the others), regardless of whether or not there were photos within the date range.

For twitter, this should apply. I need to check to see if it's broken.

>I think the intended functionality is to download any media posted later than the chosen date, however the phrasing of the option makes it seem like it could instead be to select any media posted before a certain date.

You are right. I found a bug. Will be fixed in the next version.

>I think the former is more useful but could see a benefit to being able to download all Media from a range of dates, rather than just "all before" or "all after".

Okay. I'll add it.

>It would be helpful to have this as an option in the "Download All" options as well, rather than having it only in the right-click menu for each user.

I need to think about

**Comment by AAndyProgram** [comment]
1. Without changes. 'Stories' and 'Tagged posts' will still ignore date limits.
2. **Confirmed**. Will be fixed in the next version.
3. I found a bug and changed the limit check algorithm.
4. **Added**.
5. **Will not be added!** Sorry. 'Download to date' is not a routine operation. **But** I found a bug that prevents multiple users with date restrictions from downloading. I have fixed this and in the next version you will be able to select multiple users to download within a specific date range. Examples of using this function:
    - You want to download a media block for a specific date range.
    - OR you haven't downloaded this profile for a long time and it contains too much data and you want to download some of it for the last month (for example).
    - OR you are downloading a profile for the first time and (again) only want to download data for the specified dates.
